File: src/main/java/common/world/World.java
package common.world;

import common.entities.Entity;
import talaria.common.Talaria;
import talaria.common.TalariaManager;
import talaria.common.entity.NetworkEntity;

import java.util.ArrayList;
import java.util.List;

abstract public class World extends NetworkEntity {

    public static final List<Entity> entities = new ArrayList<>();

    public World (String name, int x, int y) {}

    public void update() {
        for (Entity e : entities) {
            e.update();
            switch(TalariaManager.getSide()) {
                case CLIENT:
                    Talaria.INSTANCE.getClient().sendEntityToServer(e);
                    break;
                case SERVER:
                    Talaria.INSTANCE.getServer().sendEntityToAll(e);
                    break;
            }
        }
    }

}

File: src/main/kotlin/common/world/World.kt
package common.world

import com.github.simplenet.packet.Packet
import common.entities.Entity
import bvanseg.talaria.common.Talaria
import bvanseg.talaria.common.entity.NetworkEntity
import bvanseg.talaria.common.side.Side
import java.nio.ByteBuffer
import java.util.*

abstract class World(name: String, x: Int, y: Int) : NetworkEntity() {

    val entities: List<Entity> = ArrayList()
    private val worldSettings = WorldSettings()

    open fun update() {
        for (e in entities) {
            e.update()
            when (Talaria.getSide()) {
                Side.CLIENT -> Talaria.client!!.sendEntityToServer(e)
                Side.SERVER -> Talaria.server!!.sendEntityToAll(e)
            }
        }
    }

    override fun write(packet: Packet): Packet {
        worldSettings.write(packet)
        return packet
    }

    override fun read(buffer: ByteBuffer) {
        worldSettings.read(buffer)
    }
}
